rsconnect/main.py

```diff
@@ -408,28 +431,118 @@ def deploy_manifest(name, server, api_key, new, app_id, title, insecure, cacert,
+              help='Disable TLS certification/host validation.')
+@click.option('--cacert', '-c', envvar='CONNECT_CA_CERTIFICATE', type=click.File(),
+              help='The path to trusted TLS CA certificates.')
+@click.option('--entrypoint', '-e', help='The module and executable object which serves as the entry point for the '
+                                         'WSGi framework of choice (defaults to app:app)')
+@click.option('--exclude', '-x', multiple=True,
+              help='Specify a glob pattern for ignoring files when building the bundle. This option may be repeated/')
+@click.option('--new', '-N', is_flag=True,
+              help='Force a new deployment, even if there is saved metadata from a previous deployment. '
+                   'Cannot be used with --app-id.')
+@click.option('--app-id', '-a', help='Existing app ID or GUID to replace. Cannot be used with --new.')
+@click.option('--title', '-t', help='Title of the content (default is the same as the directory).')
+@click.option('--python', '-p', type=click.Path(exists=True),
+              help='Path to python interpreter whose environment should be used. '
+                   'The python environment must have the rsconnect package installed.')
+@click.option('--conda', '-C', is_flag=True, help='Use conda to deploy.')
+@click.option('--force-generate', '-g', is_flag=True,
+              help='Force generating "requirements.txt" or "environment.yml", even if it already exists.')
+@click.option('--verbose', '-v', is_flag=True, help='Print detailed messages.')
+@click.argument('directory', type=click.Path(exists=True, dir_okay=True, file_okay=False))
+@click.argument('extra_files', nargs=-1, type=click.Path(exists=True, dir_okay=False, file_okay=True))
+def deploy_api(name, server, api_key, insecure, cacert, entrypoint, exclude, new, app_id, title, python, conda,
+               force_generate, verbose, directory, extra_files):
+    set_verbosity(verbose)
+
+    with cli_feedback('Checking arguments'):
+        app_store = AppStore(directory)
+        connect_server = _validate_deploy_to_args(name, server, api_key, insecure, cacert)
+        entrypoint = _validate_entry_point(directory, entrypoint)
+
+        if new and app_id:
+            raise api.RSConnectException('Specify either --new/-N or --app-id/-a but not both.')
+
+        _validate_title(title)
+
+        if not entrypoint:
+            entrypoint = 'app:app'
+
+        app_id, deployment_name, title, app_mode = \
+            gather_basic_deployment_info_for_api(connect_server, app_store, directory, new, app_id, title)
+
+    click.secho('    Deploying %s to server "%s"' % (directory, connect_server.url), fg='white')
+
+    with cli_feedback('Checking server capabilities'):
+        check = [is_version_1_8_2_or_higher]
+        if conda:
+            check.append(is_conda_supported_on_server)
+        check_server_capabilities(connect_server, check)
```

Comment:
Same. It would also help if we had a server flag advertising the feature rather than checking the version.